### PR TITLE
Make metrics endpoint as optional

### PR DESCRIPTION
**Pull request by sasagarw**
This PR:
- Makes `metrics endpoint` as optional.
- fmt fixes.

/cc @periklis 
/assign @periklis 

Jira: https://issues.redhat.com/browse/LOG-1512

### CODE REVIEW

**Review by periklis — COMMENTED** [review]
LGTM

File: main.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ import (
 	"syscall"
 	"time"
 
+	metricslegacy "github.com/observatorium/api/api/metrics/legacy"
```
Comment (squat) [inline main.go:25]: Please maintain the imports ordered following the golang convention of grouping first standard library imports, then third party imports, and lastly imports from our module.

File: main.go [hunk 2]
```diff
@@ -107,6 +108,8 @@ type metricsConfig struct {
 	writeEndpoint *url.URL
 	tenantHeader  string
 	tenantLabel   string
+	// enable metrics if at least one {read|write}Endpoint} is provided.
+	enabled bool
```
Comment (squat) [inline main.go:112]: I think we need to exit with an error of neither logging nor metrics are enabled. If the administrator doesn't specify any endpoints then the API can't do anything. IMO it's the logical extrapolation of the current logic where we said "at least a metrics endpoint is mandatory". Now we would be saying "specifying at least a logging or a metrics endpoint is mandatory"

**Review by squat — CHANGES_REQUESTED** [review]
Two small details, otherwise lgtm :)

**Review by squat — APPROVED** [review]
lgtm! thanks @sasagarw

**Pull request merged**